starter-code/ShoppingListVer2/app/src/main/java/ly/generalassemb/drewmahrt/shoppinglistver2/MainActivity.java

```diff
@@ -2,25 +2,64 @@
 
 import android.content.Context;
 import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
 import android.view.View;
 import android.view.ViewGroup;
 import android.widget.CursorAdapter;
 import android.widget.ListView;
 import android.widget.SimpleCursorAdapter;
 
+import java.util.ArrayList;
+import java.util.List;
+
 import ly.generalassemb.drewmahrt.shoppinglistver2.setup.DBAssetHelper;
 
 public class MainActivity extends AppCompatActivity {
+
+    RecyclerView mRecyclerView;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        mRecyclerView = (RecyclerView) findViewById(R.id.recyclerview);
+
+        LinearLayoutManager linearLayoutManager =
+                new LinearLayoutManager(this, LinearLayoutManager.VERTICAL,false);
+        mRecyclerView.setLayoutManager(linearLayoutManager);
+
         //Ignore the two lines below, they are for setup
         DBAssetHelper dbSetup = new DBAssetHelper(MainActivity.this);
         dbSetup.getReadableDatabase();
 
+        SQLiteDatabase db = dbSetup.getReadableDatabase();
+
+        Cursor cursor = db.query("SHOPPING_LIST",
+                null,
+                null,
+                null,
+                null,
+                null,
+                null);
+
+        List<String> groceryList = new ArrayList<>();
+
+        if(cursor.moveToFirst()){
```

Comment:
Your code here in lines 40-59 should be in the SQLiteOpenHelper class. This code is what performs a query and turns the result into objects. That doesn't get done here in MainActivity, since this activity doesn't have direct access to the database itself. Instead, MainActivity calls a method in the OpenHelper, which _does_ have access to the actual database and can do this work.